nova/volume/cinder.py

```diff
@@ -1081,12 +1081,21 @@ def attachment_delete(self, context, attachment_id):
                 context, '3.44', skip_version_check=True).attachments.delete(
                     attachment_id)
         except cinder_exception.ClientException as ex:
-            with excutils.save_and_reraise_exception():
+            with excutils.save_and_reraise_exception() as ctxt:
+                code = getattr(ex, 'code', None)
+                if code == 404:
```

Comment:
What if this is an nginx 404 due to a temporary misconfiguration ?

`cinder_exception.NotFound` has also the `response` property that we can introspect, to see if the answer is json at least.

Reply:
afaik, we don't use Nginx between Nova and Cinder, because it's the internal interface. but yeah, we could look if the attachment-id is mentioned in the response to play it safe.

Reply:
But I think the internal endpoints are not exposed via ingress, so we should never reach this edge case anyway.

Reply:
We could still play it safe. Could be other misconfigurations which lead to this.

Reply:
I think that is excessive. There is an arbitrary number of was things can go wrong, but the responsibility of Nova ends at the interface. If the response is wrong, then we need to fix it there.

But looking at that again, the status should be `410 Gone`, and we might want to handle that too.

Reply:
Where did you find `410 Gone`?

Reply:
I don't think it is implemented that way, but the standard (proposal?) has reserved [410 for that case](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/410).

Finally a use-case for the soft-deleted records.